merlin-server/sql/merlin/views/resource_profile.sql

```diff
@@ -0,0 +1,51 @@
+create view resource_profile as
+(
+select profile_segment.dataset_id as dataset_id,
+       profile_segment.profile_id as profile_id,
+       profile_segment.start_offset as start_offset,
+       profile_segment.dynamics as dynamics,
+       plan.start_time + profile_segment.start_offset as start_time,
+       coalesce(
+             plan.start_time + (
+             select p.start_offset
+             from profile_segment p
+             where p.start_offset > profile_segment.start_offset
+             limit 1
+           ),
+             plan.start_time + plan.duration
+         ) as end_time
+
+from profile_segment
+       join dataset on profile_segment.dataset_id = dataset.id
+       left join plan_dataset pd on dataset.id = pd.dataset_id
+       join simulation_dataset sd on dataset.id = sd.dataset_id
```

Comment:
I think it's covered by the OR clause on line 23.

I believe the story for external profiles is that they are associated with a plan rather than a simulation, so the line `join plan on s.plan_id = plan.id OR plan.id = pd.plan_id` covers it with the `plan.id = pd.plan_id`.

I do have a moment of pause as to whether we need to do a left join on the simulation_data though.

Reply:
Agreed with your logic, as well as your pause :+1:

Reply:
Assuming it went okay in your testing, I'll resolve this conversation. Treating plan_dataset and simulation_dataset the same way using `left join` sounds like a good idea to me.

Reply:
As I was handling this one, I also noticed that we are missing a clause to filter for the profile id and dataset id on line 12